.github/workflows/test.yml

```
@@ -0,0 +1,93 @@
+name: test
+on: [pull_request, push]
+
+jobs:
+  run_tests:
+    runs-on: python:3.8
```

Maybe you want
```suggestion
    runs-on: [self-hosted]
    container: python:3.8
```
on all of them